Remove Semantic UI and replace with HTML, Tailwind and react-select

tekul opened the pull request:
Semantic UI React is unnecessarily heavyweight and opaque to work with - most of the layout components it provides can easily be replicated using standard HTML with modern CSS `flexbox` or `grid`. Components such as datepickers  are unnecessary (`<input type='date'` works just as well). This PR removes Semantic UI and replaces it with HTML and [Tailwind](https://tailwindcss.com/) for CSS and [`react-select`](https://react-select.com/home) for the  select/multiselect components which the site uses.

tekul committed:
Setup client to use tailwind CSS

The aim is to replace most (or all of the use of Semantic UI) with
Tailwind, avoiding the heavyweight components and opaque CSS that
are used everywhere and replacing them with plain HTML, styled with
TW classes.

Some specialized components may be required (datepickers, multiselects)
but these can either remain as they are or be replaced with other
options.

tekul committed:
Re-style index.js

Removed Semantic-UI components. Retained the semantic-ui css file
as it is still required by other parts of the app.

tekul committed:
Restyle Footer without semantic-ui

tekul committed:
Remove Semantic UI elements from login page

Restyle using Tailwind.

tekul committed:
Restyle Profiles page

- Replace semantic-ui with a plain HTML table
- Replace semantic-ui selects with react-select component
- Replace semantic-ui Search, Button, Grid, Divider, Message etc.

tekul committed:
Remove semantic-ui from About.js page

tekul committed:
Remove "New Profile" on global menu

There is a "New Profile" button on the dashboard so it is confusing
to have two on the page at once.

tekul committed:
Restyle ViewProfile and delete unused CSS files

- Remove semantic-ui and use a TailwindUI modal
- Remove CSS files for this page and also Profiles.js and Login.js

tekul committed:
Restyle AddNewProfile.js and remove semantic-ui deps

- Restyled the new profile form without semantic-ui.
- Removed semantic-ui.css and packages from package.json and made
some other minor styling tweaks as a result.

tekul committed:
Refactor some verbose Tailwind stuff into components

Created some internal components in the Profiles view to reduce
duplication and make tinkering with the design easier.

tekul committed:
Fix build to work properly with Tailwind and PostCSS

The production build was broken because index.css wasn't created
by `npm build`. We use the instructions for setting up Tailwind
to work with React.

Uses craco in place of react-scripts to allow overriding of PostCSS
configuration. Tailwind needs PostCSS 8 which the current react-scripts
doesn't yet support so the PostCSS 7 compatibilty version has been
installed for now. This should be updated when react is compatible.

Packages such as Tailwinds apparently need to be in the 'dependencies'
section in package.json as the production client is build by the
heroku-postbuild command which ignores dev dependencies.

kabaros commented on client/src/components/AddNewProfile.js:
```diff
@@ -28,7 +27,7 @@ const AddNewProfile = () => {
     updateField(event.target.name, event.target.value);
   };
-
+  console.log(profileData);
```
remove console.log

kabaros commented on client/src/components/AddNewProfile.js:
```diff
@@ -96,187 +86,202 @@ const AddNewProfile = () => {
-      <Form onSubmit={createProfile}>
-        <ul class="errors">
+    <div className="px-10 max-w-4xl mx-auto">
+      <form onSubmit={createProfile}>
```
if there is too much forms and validation work in the app, then I would look at using [formik](https://formik.org/) and [yup](https://github.com/jquense/yup)

tekul replied:
Thanks for the suggestions. The current validation here isn't really adequate so it'll be a task to improve it.

kabaros commented on client/src/components/Profiles.js:
```diff
@@ -128,153 +110,216 @@ const Profiles = () => {
+
+const Td = ({ children }) => <td className="px-5 py-3">{children}</td>;
+
+const Search = props => (
```
maybe consider moving some of these components (at least Search) to new file .. unless you think they will only live within the Profiles component

tekul replied:
Yeah, for now they are only used here so I decided to keep them internal to this page.

kabaros commented on client/src/components/ViewProfile.js:
```diff
@@ -35,150 +22,197 @@ const ViewProfile = () => {
     }
   }, [profileId, delProfile, history]);
 
+  const confirmDelete = deleteOrNot => {
```
`deleteOrNot` feels like a weird name .. not sure what true would mean in this case. Maybe `deleteConfirmed` is clearer.

kabaros commented on client/src/index.js:
```diff
@@ -27,57 +26,43 @@ const Routes = () => {
-            <List.Item name="logout">
-              <Button onClick={logout}>Logout </Button>
-            </List.Item>
+            <button onClick={logout} className="nav-btn">
```
you can use the value for `isLoggedIn` here instead of getting the token from localStorage again?

tekul replied:
I noticed this but left it for now since it's not really relevant to these changes. It can be picked up in a task.

kabaros reviewed (dismissed):
it looks good. Some small comments/opinions but it all looks good and standard to me.

tekul committed:
Changes based on PR feedback from kabaros

- Remove debugging console.log call
- Rename function parameter for delete confirmation modal

costanzazanoletti commented on client/src/components/ViewProfile.js:
```diff
@@ -35,150 +22,197 @@ const ViewProfile = () => {
+        <div>
+          <div className="flex justify-left items-center mt-10 space-x-5">
+            <h1 className="text-2xl font-bold">{`${profile.first_name} ${profile.last_name}`}</h1>
+            <div className="text-lg">{profile.status}</div>
```
For the status I would keep the same layout as the profile list, with different color depending on the status.  
Maybe we can move the function getClassName to an helper file so that it's available for both components.

tekul replied:
Yes, it would be better if it used the same colour codes. There are a lot of issues like that though (the original version used the yellow colour for all the profiles). This PR was mainly about getting rid of Semantic UI so where other stuff was broken I just left it in a basic state rather than trying to fix everything. Some of the other information in the profile view doesn't show at all (like the nationality), so there is plenty of scope for improvement.

costanzazanoletti reviewed (approved):
It looks much cleaner than with semantic-ui, I added just a comment on the ViewProfile component
